clients/tabby-agent/src/codeCompletion/contexts.ts

```diff
@@ -76,25 +83,38 @@ export class CompletionContext {
   mode: "default" | "fill-in-line";
   hash: string;
 
+  // example of auto complete part
+  // cons| -> console
+  // completionItem: console
+  // insertPosition: 4
+  // insertSeg: ole
+  // currSeg: cons
+  completionItem: string = "";
+  insertPosition: number = 0;
+  insertSeg: string = "";
+  currSeg: string = "";
+  withCorrectCompletionItem: boolean = false; // weather we are using completionItem or not
```

Comment:
I think using data structure like `SelectedCompletionInfo` directly should be fine.

Reply:
Offline update: Retain the current data structure for future PR optimizations related to fetch code quality.